### PR TITLE
universal parsing now integrated into EKR

### PR DESCRIPTION
**Pull request by snova-kwasia**

### CODE REVIEW

**Comment by snova-jorgep** [comment]
@snova-kwasia  for universal parsing in EKR, everything looks, tested, and is working well, we need to update the EKR readme  to include instructions to launch the universal parser docker container, or alternatively local env installation

File: utils/eval/rag_eval.py [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+    context_precision,
+    answer_correctness,
+    answer_similarity,
+    context_recall,
+    context_relevancy,
+)
+from ragas import evaluate
+import wandb
+import argparse
+from langchain_community.llms.sambanova import SambaStudio, Sambaverse
```
Comment (snova-jorgep) [inline utils/eval/rag_eval.py:20]: those looks like are not being used, where are the models being instanced?

File: eval_jumpstart/README.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+Answer Relevancy
+Answer Correctness
+Answer Similarity
+Context Precision
+Context Recall
+Faithfulness
+Context Relevancy
+
+These are from RAGAS please see the RAGAS website for details on the underlying methodlogy
```
Comment (snova-jorgep) [inline eval_jumpstart/README.md:132]: typo in methodology

File: eval_jumpstart/README.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+evaluation:
+  num_samples: 100
+  log_wandb: true
+  project_name: "rag-eval"
+  eval_name: "rag_eval_test"
+```
+
+
+# Evaluation Use Cases
```
Comment (snova-jorgep) [inline eval_jumpstart/README.md:53]: Think Worth it to include a short explanation of each use case

File: utils/eval/config.yaml [hunk 4]
```diff
@@ -0,0 +1,69 @@
+eval_dataset:
+  path: eval_set.csv
+  # hf_dataset_name: squad_v2
+  question_col: question
+  ground_truth_col: ground_truth
+  answer_col: answer
+  #context_col: contexts
```
Comment (snova-jorgep) [inline utils/eval/config.yaml:7]: why is this commented, saw streamlit app requires it

File: utils/eval/app.py
Comment (snova-jorgep) [inline utils/eval/app.py:1]: Should we include the st app usage in the readme?

**Review by snova-jorgep — CHANGES_REQUESTED** [review]

File: enterprise_knowledge_retriever/src/document_retrieval.py [hunk 5]
```diff
@@ -172,155 +180,48 @@ def set_llm(self):
     def get_pdf_text_and_metadata_pypdf2(self, pdf_doc, extra_tags=None):
         """Extract text and metadata from pdf document with pypdf2 loader
```
[the remark this answers is not in the document]
Comment (snova-kwasia) [reply enterprise_knowledge_retriever/src/document_retrieval.py:181]: fixed

File: utils/parsing/README.md
[the remark this answers is not in the document]
Comment (snova-kwasia) [reply utils/parsing/README.md:1]: fixed

File: utils/parsing/README.md [hunk 6]
```diff
@@ -34,14 +34,14 @@ Before using SambaParse, make sure you have the following:
    pip install git+https://github.com/sambanova/ai-starter-kit
```
[the remark this answers is not in the document]
Comment (snova-kwasia) [reply utils/parsing/README.md:34]: WIll update the reqs.

File: utils/parsing/docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    driver: bridge
+
+services:
+  unstructured-api:
+    image: downloads.unstructured.io/unstructured-io/unstructured-api:latest
+    command: --port 8000 --host 0.0.0.0
+    ports:
+      - "${UNSTRUCTURED_PORT:-8005}:8000"
+    env_file:
+      - ./.env
```
[the remark this answers is not in the document]
Comment (snova-kwasia) [reply utils/parsing/docker-compose.yml:14]: updated

File: enterprise_knowledge_retriever/streamlit/app.py [hunk 8]
```diff
@@ -101,9 +101,9 @@ def main():
             if st.button("Process"):
                 with st.spinner("Processing"):
                     # get pdf text
-                    raw_text, meta_data = documentRetrieval.get_data_for_splitting(docs)
+                    text_chunks = documentRetrieval.parse_doc(docs)
                     # get the text chunks
-                    text_chunks = documentRetrieval.get_text_chunks_with_metadata(docs=raw_text, meta_data=meta_data)
+                    #text_chunks = documentRetrieval.get_text_chunks_with_metadata(docs=raw_text, meta_data=meta_data)
```
[the remark this answers is not in the document]
Comment (snova-kwasia) [reply enterprise_knowledge_retriever/streamlit/app.py:106]: done

**Review by snova-jorgep — APPROVED** [review]

**Pull request merged**